### DISCUSSION TITLE
setup.sh with LDAP configuration

### DISCUSSION THREAD

**Discussion by cociweb** [docker-mailserver/docker-mailserver#2464] [Q&A] [answered]
Hello,

I try to setup this awesome server. I would liket to setup the accounts. I've configured ldap (I hope coorectly) but now I'm stucked. I don't know what is the next step. I would like to create users/mailboxes from scretch. (without ldap I can succesfully create them with setup.sh.) With the ldap config the setup.sh gets into an endless loop, and no account is created. the logs are useless. 
```
# ./setup.sh email add user@example.org BreakMe
Waiting for dovecot to create /path/to/mailbox/example.org/user...
Waiting for dovecot to create /path/to/mailbox/example.org/user...
....
```
After interuption:
`^X^C[[  INF  ]]  Removed lock /tmp/docker-mailserver/addmailuser.lock.`
(not posssible to add it again, until it is not deleted:

```
# ./setup email del user@example.org
Do you want to delete the mailbox as well (removing all mails) ? [Y/n] y
Error: sed -i -e /^user@example\.org:.*$/d /tmp/docker-mailserver/dovecot-quotas.cf
Quota for user@example.org couldn't be deleted in /tmp/docker-mailserver/dovecot-quotas.cf.
Mailbox directory '/path/to/mailboxl/example.org/user' did not exist.
Error :: See the messages above.
Aborting.
[[  INF  ]]  Removed lock /tmp/docker-mailserver/delmailuser.lock.
```

Should the user be ready in ldap before executing setup.sh? I've searched in the docs, but there is no clue, how to create mailbox, if ldap is used. What is the correct order? if it is not possible with setup.sh, what is the correct manual steps and order  to make it working with ldap?

Thank's for your hints.

**Comment by cociweb** [comment]
I'm afraid, [changchichung](https://github.com/changchichung) has the same issue here: https://github.com/docker-mailserver/docker-mailserver/issues/2406, but there is no clarification, what are the steps, what He/I should take.

**Comment by casperklein** [comment, marked as answer]
See Georgs  answer: https://github.com/docker-mailserver/docker-mailserver/issues/2406#issuecomment-1036020290

Until someone make it work and provides the solution as PR, it's not possible with DMS to use LDAP and local accounts together.
